starter-code/routes/passportRouter.js

```diff
@@ -1,11 +1,44 @@
 const express        = require("express");
 const passportRouter = express.Router();
+const bcrypt  = require('bcrypt');
 // Require user model
+const User = require('../models/user')
+const passport = require('passport')
 
 // Add bcrypt to encrypt passwords
 
-// Add passport 
+//SIGNUP
+passportRouter.get('/signup', (req, res, next) => {
+  res.render('passport/signup')
+})
 
+// Signup passport 
+passportRouter.post('/signup', (req, res, next) => {
+  const { username, password } = req.body
+
+  bcrypt.hash(password, 10)
+    .then(hash => {
+      return User.create({
+        username: username,
+        password: hash
+      })
+      .then(user => {
+        res.send(user)
+      })
+    })
+})
+
+//LOGIN
+passportRouter.get('/login', (req, res, next) => {
+  res.render('passport/login')
+})
+
+//Login passport
+passportRouter.post('/login', passport.authenticate('local', {
+  successRedirect: '/',
+  failureRedirect: '/auth/login',
+  failureFlash: true
+}))
 
 const ensureLogin = require("connect-ensure-login");
```

Comment:
Small note: it's best practice to have all your required modules at the top of the file.